### PR TITLE
Added Word2Vec in Tensorflow

### PR DESCRIPTION
**Pull request by aashishpiitk**

### CODE REVIEW

File: NLP/word2vec/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Tensorflow Implementation of Word2Vec (Dataset from kaggle file)
+
+## Usage
+### To train
```
Comment (nirmal-suthar) [inline NLP/word2vec/README.md:4]: you can also add help log for indicating available arguments

File: NLP/word2vec/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+* [Stanford Word2Vec Notes](http://web.stanford.edu/class/cs224n/readings/cs224n-2019-notes01-wordvecs1.pdf)
+
+* [Original Word2Vec paper](http://arxiv.org/pdf/1301.3781.pdf)
+
+* [Google Word2vec paper which suggested improvements in training using negative sampling and sub-sampling](http://papers.nips.cc/paper/5021-distributed-representations-of-words-and-phrases-and-their-compositionality.pdf)
+
+## Contributed by:
+* [Aashish Patel](https://github.com/aashishpiitk/)
+
+# Summary
```
Comment (nirmal-suthar) [inline NLP/word2vec/README.md:35]: Optionally, show some results (maybe in form example in readme )

File: NLP/word2vec/dataloader.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+from nltk.corpus import stopwords
+import nltk
+nltk.download('stopwords')
+
+# Downloading the dataset
+# !kaggle datasets download -d harmanpreet93/hotelreviews
+# unzip the dataset and keep it in a folder named hotelreviews
+
+
+
```
Comment (nirmal-suthar) [inline NLP/word2vec/dataloader.py:12]: add this instruction in readme

File: NLP/word2vec/evaluation.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    topTen = list()
+    for word in vocab:
+        topTen.append([word, getSimilarityByEmbedding(word4_emb,emb[word_to_id[word]])])
+    topTen.sort(key = lambda x: x[1],reverse=True)
+    return topTen[:10] 
+
+
+
+
```
Comment (nirmal-suthar) [inline NLP/word2vec/evaluation.py]: remove stray lines

File: NLP/word2vec/main.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                    metrics= ['accuracy'])
+        model.fit(oneHotNumpy[:,0,:],oneHotNumpy[:,1,:],
+                epochs = epochs,
+                batch_size = batch_size)
+
+
+        emb = model.get_weights()[0]
+        save("word2vec_embeddings.npy",emb)
+
+
```
Comment (nirmal-suthar) [inline NLP/word2vec/main.py]: stray lines

File: NLP/word2vec/utils.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  data = {
+      "vec":vec,
+      "word_to_id":word_to_id,
+      "id_to_word":id_to_word,
+      "vocabSize":vocabSize,
+      "vocab":vocab
+  }
+  return data
+
+
```
Comment (nirmal-suthar) [inline NLP/word2vec/utils.py]: stray lines

**Review by nirmal-suthar — APPROVED** [review]

File: NLP/word2vec/dataloader.py (hunk 3 above)
Comment (aashishpiitk) [reply NLP/word2vec/dataloader.py:12]: done

File: NLP/word2vec/evaluation.py (hunk 4 above)
Comment (aashishpiitk) [reply NLP/word2vec/evaluation.py]: done

File: NLP/word2vec/main.py (hunk 5 above)
Comment (aashishpiitk) [reply NLP/word2vec/main.py]: done

File: NLP/word2vec/utils.py (hunk 6 above)
Comment (aashishpiitk) [reply NLP/word2vec/utils.py]: done

File: NLP/word2vec/README.md (hunk 1 above)
Comment (aashishpiitk) [reply NLP/word2vec/README.md:4]: is it necessary

Comment (aashishpiitk) [reply NLP/word2vec/README.md:4]: done

**Comment by dev-chauhan** [comment]
It is better not to push *.pyc files on GitHub, you can use `.gitignore`. It makes review of your code harder.

**Review by dev-chauhan — APPROVED** [review]

**Comment by nirmal-suthar** [comment]
> It is better not to push *.pyc files on GitHub, you can use `.gitignore`. It makes review of your code harder.

yes remove __pycache__ folder

**Comment by aashishpiitk** [comment]
> > It is better not to push *.pyc files on GitHub, you can use `.gitignore`. It makes review of your code harder.
> 
> yes remove **pycache** folder

done

**Comment by namangup** [comment]
@aashishpiitk I looked at the code and it seems that you are using keras instead of tensorflow.keras .

**Pull request merged**